crypto/src/main/java/org/hyperledger/besu/crypto/BouncyCastleSecurityModule.java

```diff
@@ -15,32 +15,42 @@
 package org.hyperledger.besu.crypto;
 
 import org.hyperledger.besu.crypto.SECP256K1.KeyPair;
+import org.hyperledger.besu.plugin.services.securitymodule.PublicKey;
+import org.hyperledger.besu.plugin.services.securitymodule.SecurityModule;
+import org.hyperledger.besu.plugin.services.securitymodule.Signature;
 
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
 import org.apache.tuweni.bytes.Bytes32;
 
+/**
+ * Implementation of SecurityModule deferring cryptographic operations via SECP256K1 class which
+ * uses BouncyCastle.
+ */
 public class BouncyCastleSecurityModule implements SecurityModule {
 
-  private final KeyPair nodeKeys;
+  private final Supplier<KeyPair> keyPairSupplier;
```

Comment:
why the change to a Supplier? This _feels_ like a change which shouldn't be part of this PR.

Reply:
Its because of plugin lifecycle. `BouncyCastleCliOptions.getPrivateKeyFile();` is not populated during the register command. (as we register PicoCli service in register method as well). Hence we need to provide a method reference which allows the actual cal to be deferred while all cli options are initialized.

Reply:
Other plugins are also following similar approach where they want to utilize cli options.

Reply:
Reverted and fixed AbstractCommandTest which resulted in this change.